Sources/SwiftParser/TokenPrecedence.swift

```diff
@@ -192,47 +192,266 @@ enum TokenPrecedence: Comparable {
     case  // Literals
     .Self, .false, .nil, .`self`, .super, .true:
       self = .identifierLike
+
     // MARK: Expr keyword
     case  // Keywords
-    .as, .is, .try,
+    .as, .is, .some, .try,
+      .await, .reasync, .async,
       // We don't know much about which contextual keyword it is, be conservative an allow considering it as unexpected.
       // Keywords in function types (we should be allowed to skip them inside parenthesis)
       .rethrows, .throws,
       // Consider 'any' a prefix operator to a type and a type is expression-like.
       .Any,
       // 'where' can only occur in the signature of declarations. Consider the signature expression-like.
       .where,
+      .each,
       // 'in' occurs in closure input/output definitions and for loops. Consider both constructs expression-like.
       .in:
       self = .exprKeyword
+
     case  // Control-flow constructs
     .defer, .do, .for, .guard, .if, .repeat, .switch, .while,
       // Secondary parts of control-flow constructs
       .case, .catch, .default, .else,
       // Return-like statements
       .break, .continue, .fallthrough, .return, .throw, .then, .yield:
       self = .stmtKeyword
+
     // MARK: Decl keywords
     case  // Types
-    .associatedtype, .class, .enum, .extension, .protocol, .struct, .typealias,
+    .associatedtype, .class, .enum, .extension, .protocol, .struct, .typealias, .actor, .Sendable,
       // Access modifiers
       .fileprivate, .internal, .private, .public, .static,
       // Functions
       .deinit, .func, .`init`, .subscript,
       // Variables
-      .let, .var,
+      .let, .var, .weak,
```

Comment:
`weak` should be an `exprKeyword` for now since we can use it in closure capture items and in that context we are more in an expression-style context than a context that introduces new declarations.